src/components/DatePicker/components/Day/Day.tsx

```diff
@@ -83,7 +87,7 @@ export class Day extends React.PureComponent<CombinedProps, never> {
         aria-disabled={disabled}
         role="gridcell"
       >
-        {date}
+        {Intl.DateTimeFormat(locale, {day: 'numeric'}).format(day)}
```

Comment:
Based on this Slack convo:
![image](https://user-images.githubusercontent.com/445045/52639415-3c22bd00-2ea2-11e9-8263-863f8701942f.png)

We should keep `date` for now.